panic: runtime error: invalid memory address or nil pointer dereference

### What happened

after I tried to use `tbls doc` with postgre, the error occurred.
<img width="820" alt="截屏2020-08-11 23 45 11" src="https://user-images.githubusercontent.com/69157011/89912045-15774b00-dc2d-11ea-9768-729b7bc02def.png">

### What you expected to happed

get this fixed.

### What stacktrace or error message from tbls did you see?

here it is:

`tbls doc`
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41b8fb2]

goroutine 1 [running]:
database/sql.(*Rows).close(0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/k1low/.anyenv/envs/goenv/versions/1.13.10/src/database/sql/sql.go:3063 +0x72
database/sql.(*Rows).Close(0x0, 0x5109c4c, 0x3d0)
	/Users/k1low/.anyenv/envs/goenv/versions/1.13.10/src/database/sql/sql.go:3059 +0x33
github.com/k1LoW/tbls/drivers/postgres.(*Postgres).Analyze(0xc00039da80, 0xc000377570, 0x5399d20, 0xc000449ee0)
	/Users/k1low/src/github.com/k1LoW/tbls/drivers/postgres/postgres.go:141 +0x28bb
github.com/k1LoW/tbls/datasource.Analyze(0xc000036640, 0x49, 0x0, 0x0, 0x0, 0x0)
	/Users/k1low/src/github.com/k1LoW/tbls/datasource/datasource.go:111 +0x80b
github.com/k1LoW/tbls/cmd.glob..func5(0x5cfd1e0, 0x5d7c5a8, 0x0, 0x0)
	/Users/k1low/src/github.com/k1LoW/tbls/cmd/doc.go:73 +0x1b0
github.com/spf13/cobra.(*Command).execute(0x5cfd1e0, 0x5d7c5a8, 0x0, 0x0, 0x5cfd1e0, 0x5d7c5a8)
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:846 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x5cfd9c0, 0x4, 0x5d7c5a8, 0x0)
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:887
github.com/k1LoW/tbls/cmd.Execute()
	/Users/k1low/src/github.com/k1LoW/tbls/cmd/root.go:184 +0x86
main.main()
	/Users/k1low/src/github.com/k1LoW/tbls/main.go:33 +0x20

### Anything else we need to know?
I get the same error when using other postgresql databases.

### Environment

- `tbls version`
1.43.0

- Database version
psql (PostgreSQL) 9.2.24

- OS 
`macOS Calalina`

- Shell 
`zsh`

- Stacktrace or error message from tbls ( If you can, set `DEBUG=1`)
`DEBUG=1 tbls doc`
and the same error message.

- Config ( .tbls.yml )
dsn: postgres://user:pw@host:5432/db?sslmode=disable

--- a/datasource/datasource_test.go
+++ b/datasource/datasource_test.go
@@ -68,7 +68,7 @@ func TestAnalyzeTables(t *testing.T) {
 		want := tt.tableCount
 		got := len(schema.Tables)
 		if got != want {
-			t.Errorf("got %v\nwant %v", got, want)
+			t.Errorf("%v: got %v\nwant %v", tt.dsn, got, want)
 		}
 	}
 }

--- a/drivers/postgres/postgres.go
+++ b/drivers/postgres/postgres.go
@@ -149,8 +149,8 @@ ORDER BY oid`)
 				constraintDef                  string
 				constraintType                 string
 				constraintReferenceTable       sql.NullString
-				constraintColumnNames          []string
-				constraintReferenceColumnNames []string
+				constraintColumnNames          []sql.NullString
+				constraintReferenceColumnNames []sql.NullString
 				constraintComment              sql.NullString
 			)
 			err = constraintRows.Scan(&constraintName, &constraintDef, &constraintType, &constraintReferenceTable, pq.Array(&constraintColumnNames), pq.Array(&constraintReferenceColumnNames), &constraintComment)
@@ -163,9 +163,9 @@ ORDER BY oid`)
 				Type:             convertConstraintType(constraintType),
 				Def:              constraintDef,
 				Table:            &table.Name,
-				Columns:          constraintColumnNames,
+				Columns:          arrayRemoveNull(constraintColumnNames),
 				ReferenceTable:   &rt,
-				ReferenceColumns: constraintReferenceColumnNames,
+				ReferenceColumns: arrayRemoveNull(constraintReferenceColumnNames),
 				Comment:          constraintComment.String,
 			}
 
@@ -279,7 +279,7 @@ ORDER BY attr.attnum;
 			var (
 				indexName        string
 				indexDef         string
-				indexColumnNames []string
+				indexColumnNames []sql.NullString
 				indexComment     sql.NullString
 			)
 			err = indexRows.Scan(&indexName, &indexDef, pq.Array(&indexColumnNames), &indexComment)
@@ -290,7 +290,7 @@ ORDER BY attr.attnum;
 				Name:    indexName,
 				Def:     indexDef,
 				Table:   &table.Name,
-				Columns: indexColumnNames,
+				Columns: arrayRemoveNull(indexColumnNames),
 				Comment: indexComment.String,
 			}
 
@@ -393,8 +393,8 @@ SELECT
   END AS def,
   cons.contype AS type,
   fcls.relname,
-  ARRAY_REMOVE(ARRAY_AGG(attr.attname), NULL),
-  ARRAY_REMOVE(ARRAY_AGG(fattr.attname), NULL),
+  ARRAY_AGG(attr.attname),
+  ARRAY_AGG(fattr.attname),
   descr.description AS comment
 FROM pg_constraint AS cons
 LEFT JOIN pg_trigger AS trig ON trig.tgconstraint = cons.oid AND NOT trig.tgisinternal
@@ -408,6 +408,17 @@ AND (cons.conkey IS NULL OR attr.attnum = ANY(cons.conkey))
 AND (cons.confkey IS NULL OR fattr.attnum = ANY(cons.confkey))
 GROUP BY cons.conindid, cons.conname, cons.contype, cons.oid, trig.oid, fcls.relname, descr.description
 ORDER BY cons.conindid, cons.conname`
+}
+
+// arrayRemoveNull
+func arrayRemoveNull(in []sql.NullString) []string {
+	out := []string{}
+	for _, i := range in {
+		if i.Valid {
+			out = append(out, i.String)
+		}
+	}
+	return out
 }
 
 func (p *Postgres) queryForIndexes() string {
@@ -427,7 +438,7 @@ ORDER BY idx.indexrelid`
 SELECT
   cls.relname AS indexname,
   pg_get_indexdef(idx.indexrelid) AS indexdef,
-  ARRAY_REMOVE(ARRAY_AGG(attr.attname), NULL),
+  ARRAY_AGG(attr.attname),
   descr.description AS comment
 FROM pg_index AS idx
 INNER JOIN pg_class AS cls ON idx.indexrelid = cls.oid

